rpc/v8/storage.go

```diff
@@ -163,24 +163,34 @@ func processStorageKeys(storageKeys []StorageKeys) ([]StorageKeys, *jsonrpc.Erro
 
 // isBlockSupported checks if the block ID requested is supported for storage proofs
 // Currently returns true only if the block ID requested matches the head block
-func isBlockSupported(id BlockID, head *core.Block) *jsonrpc.Error {
+func (h *Handler) isBlockSupported(id BlockID, head *core.Block) *jsonrpc.Error {
+	var blockNumber uint64
 	switch {
 	case id.IsLatest():
 		return nil
 	case id.IsPending():
 		// TODO: Remove this case when specs replaced BLOCK_ID by another type.
 		return rpccore.ErrCallOnPending
 	case id.GetHash() != nil:
-		if id.GetHash().Equal(head.Hash) {
-			return nil
+		header, err := h.bcReader.BlockHeaderByHash(id.GetHash())
+		if err != nil {
+			if errors.Is(err, db.ErrKeyNotFound) {
+				return rpccore.ErrBlockNotFound
+			}
+			return rpccore.ErrInternal.CloneWithData(err)
 		}
-		return rpccore.ErrStorageProofNotSupported
+		blockNumber = header.Number
```

Comment:
a small question out of curiosity, is the header.Number any different than BlockID.Number? I'm trying to understand, why do we have to get the information from the DB